src/Dev/Deprecation.php

```diff
@@ -353,4 +451,9 @@ public static function restore_settings($settings)
         static::notice('4.12.0', 'Will be removed without equivalent functionality to replace it');
         // noop
     }
+
+    private static function envVarIsSet(string $var, $val)
+    {
+        $val !== null || !($val === false && getenv($var) === false);
```

Comment:
I thought it would always return `null` if it wasn't set, but `Environment::getEnv()` falls back on `getenv()` which returns `false` if it can't get a value (i.e. a value is not set)

Reply:
This method doesn't have a return value?

Given that it's called a bunch, you'll probably want to check this PR still works as expected once you start returning things

Reply:
Oops! Though now that I look at the PHPDoc for `Environment::getEnv()` it explicitly states that `false` is returned from that if the variable isn't set, so I've removed this method, I was over-engineering it.
In any case I've added tests to check that the behaviour works as expected and overrides when an env var is set but uses config otherwise.